### ISSUE TITLE
ambiguous implicit when "-Ypartial-unification" enable in Scala 2.12

### ISSUE THREAD

**Issue by xuwei-k** [scalaz/scalaz#1286]
```
tests/src/test/scala/scalaz/MaybeTTest.scala:34: ambiguous implicit values:
[error]  both method monadMTMAB in object Monad of type [MT[_[_], _], MAB[_, _], A](implicit t: scalaz.MonadTrans[MT], implicit m: scalaz.Monad[[β$0$]MAB[A,β$0$]])scalaz.Monad[[β$2$]MT[[β$1$]MAB[A,β$1$],β$2$]]
[error]  and method maybeTMonadPlus in class MaybeTInstances0 of type [F[_]](implicit F0: scalaz.Monad[F])scalaz.MonadPlus[[β$4$]scalaz.MaybeT[F,β$4$]]
[error]  match expected type scalaz.Monad[[β$11$]scalaz.MaybeT[[β$12$]scalaz.\/[E,β$12$],β$11$]]
[error]     def monadEither[E] = Monad[MaybeT[E \/ ?, ?]]
[error]                               ^
```

- https://github.com/scalaz/scalaz/blob/v7.3.0-M6/tests/src/test/scala/scalaz/MaybeTTest.scala#L34
- https://github.com/scalaz/scalaz/blob/v7.3.0-M6/core/src/main/scala/scalaz/Monad.scala#L92
- https://github.com/scalaz/scalaz/commit/5c34aa0e9d24f4962eaf5b7b1850fa7cee377747#diff-4c4dbf6daa27ff3ce049300b6febead9R92
- https://github.com/scalaz/scalaz/blob/v7.3.0-M6/core/src/main/scala/scalaz/MaybeT.scala#L114

**Comment by TomasMikula** [comment]
This causes problems even without `-Ypartial-unification`. It is reproducible in scalaz's sbt's console on Scala 2.10:

```
> testsJVM/console
[info] Starting scala interpreter...
[info] 
Welcome to Scala version 2.10.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_102).
Type in expressions to have them evaluated.
Type :help for more information.

scala> :paste
// Entering paste mode (ctrl-D to finish)

import scalaz._
import scalaz.std.string._

type IntStateT[F[_], A] = StateT[F, Int, A]

type Foo[A] = Writer[String, A]
type Bar[A] = IntStateT[Foo, A]
Monad[Bar]

// Exiting paste mode, now interpreting.

<console>:16: error: ambiguous implicit values:
 both method stateTMonadState in class StateTInstances2 of type [S, F[_]](implicit F0: scalaz.Monad[F])scalaz.MonadState[[γ$13$]scalaz.IndexedStateT[F,S,S,γ$13$],S]
 and method monadMTMAB in object Monad of type [MT[_[_], _], MAB[_, _], A](implicit t: scalaz.MonadTrans[MT], implicit m: scalaz.Monad[[β$0$]MAB[A,β$0$]])scalaz.Monad[[β$2$]MT[[β$1$]MAB[A,β$1$],β$2$]]
 match expected type scalaz.Monad[Bar]
              Monad[Bar]
                   ^

scala> 
```

**Comment by TomasMikula** [comment]
I would just remove [monadMTMAB](https://github.com/scalaz/scalaz/blob/v7.3.0-M6/core/src/main/scala/scalaz/Monad.scala#L92) and enable `-Ypartial-unification`.

**Comment by xuwei-k** [comment]
@jedws Is there any opinions? You added `monadMTMAB` https://github.com/scalaz/scalaz/commit/5c34aa0e9d24f4962eaf5b7b1850fa7cee377747#diff-4c4dbf6daa27ff3ce049300b6febead9R92

**Comment by TomasMikula** [comment]
With `-Ypartial-unification` now in 2.11.11, should 7.3.x series drop support for Scala 2.10 and enable `-Ypartial-unification`?

**Comment by xuwei-k** [comment]
> 7.3.x series drop support for Scala 2.10 and enable -Ypartial-unification?

👍

**Comment by TomasMikula** [comment]
Resolved by #1367
